### DISCUSSION TITLE
Android - Full Network Access because of AIR injected android.permission.INTERNET

### DISCUSSION THREAD

**Discussion by teotigraphix** [airsdk/Adobe-Runtime-Support#2469] [Q&A] [answered]
I thought I would ask about this again.

`android.permission.INTERNET`

In this day and age of paranoia/network security, is there any reason why we are forced to keep this permission publishing for Android these days?

I have a simple app that needs nothing, but Google Play says "full network access", users are saying, yeah OK.

Thoughts?

If the answer is no, "We need it", I am really considering this a major flaw with AIR Android publishing now.

EDIT: I understand this has to do with using sockets, but my app does not use sockets....

**Comment by ajwfrost** [comment, marked as answer]
I had mistakenly thought that this was just something injected if you hadn't defined your own.. but just checked the code, it seems to inject this all the time. There's a comment: "Captive runtime apps require these additional permissions, which would normally be in the runtime apk" but I'm not 100% sure what exactly might require that. Once upon a time, Adobe collected metrics from installs/first-run of AIR applications, but that was stopped when the European privacy act was published..

We can perhaps just take this out and check that normal functionality works still. It can always be added back in when needed!

If you want to remove it currently, you can generate the "AndroidStudio" project rather than going straight to APK/AAB, and then edit the manifest manually before building in Android Studio.

thanks

**Reply by teotigraphix** [reply]
Alright, thanks @ajwfrost ! I thought it might be an oversight from past implementations. 

This would be GREAT to get rid of default wise.

**Reply by marchbold** [reply]
Always wondered about this one and why it was included by default.

If you do remove it from default we'll likely need to update a lot of the airpackages and ANE documentation as we assume this one is included :)

**Reply by ajwfrost** [reply]
Sorry @marchbold  but we're removing it in the 50.2 release.. we can potentially look at an interim mechanism where we re-add it if there are certain dependencies/ANEs listed?

**Reply by marchbold** [reply]
All good, I should have been adding it anyways so will get the docs  and airpackages updated. Cheers

**Reply by balaedit** [reply]
Thanks, It's also mine requirement as my app uses the offline data only go online after the file is not available but I could not able to off the internet access only for the app. Does it help that we can control via giving button in the app that app could use the internet or not? If it is then it would would be very nice. If not please add/suggestion about this controls also.

**Reply by marchbold** [reply]
@balaedit If you need network access at all in your application you'll need to add this permission. Removing this permission will disable network access permanently for your app.

**Reply by balaedit** [reply]
Thanks @marchbold, but removing this permission will permanently disabled the network access. My requirement is that user can enable and disable the internet usage for my app using toggle button. It will not enable or disable the mobile data/wifi. It should control only for the app.

**Reply by marchbold** [reply]
@balaedit Yeah sure, this permission won't achieve that, you'll need a different approach.

**Reply by balaedit** [reply]
thanks @marchbold, I can achieve the enable/disable the mobile data/wifi data using ANE creation. But It is not fulfill the requirement. My goal is to specific to the app.

**Reply by marchbold** [reply]
@balaedit As you control your code I'd suggest this is an implementation thing rather than something you should be looking for an external solution too. As far as I'm aware there's no real solution to toggle all network access for your app, I'd suggest you implement a "network manager" and route all your network calls through it and then have it manage the state.

**Comment by teotigraphix** [comment]
@ajwfrost Thanks!

![image](https://user-images.githubusercontent.com/226417/222295991-6cc26fc5-41f0-4579-8915-88aa85aca898.png)
